src/compiler/checker.ts

```diff
@@ -14133,19 +14178,100 @@ namespace ts {
                     }
                 }
 
-                // Checks for ambient external modules.
                 if (isAmbientExternalModule) {
-                    if (!isGlobalSourceFile(node.parent)) {
-                        error(node.name, Diagnostics.Ambient_modules_cannot_be_nested_in_other_modules_or_namespaces);
+                    if (isExternalModuleAugmentation(node)) {
+                        // if symbol of augmentation is not merged this means that either
+                        // - this is an augmentation of the global scope 
+                        // or
+                        // - this augmentation was not merged with main definition of the module
+                        //   error should already be reported so all errors in the body of augmentation can be ignored.

```

Comment:
you have augmentation for some module './m' but actual definition of the module cannot be found. We've already reported an error that './m' cannot be found so there is no need to dive into the body of augmentation at for every declaration say that it is incorrect because it introduces a new name in module